### PR TITLE
update create account API

### PR DESCRIPTION
**Pull request by sadam-123**

### CODE REVIEW

File: backend/models/user.py [hunk 1]
```diff
@@ -3,35 +3,48 @@
 from typing import Optional, List
 from pydantic import BaseModel, validator, constr
 
+
+class UserRole(str, Enum):
+    ANGANVADI = 'ANGANVADI'
```
Comment (sumit-rawat-quo) [inline backend/models/user.py]: Use "ANGANWADI" instead

File: backend/models/user.py [hunk 2]
```diff
@@ -3,35 +3,48 @@
...
 
+
 class CreateUserSchema(BaseModel):
     name: str
     phone_number: constr(
         strip_whitespace=True,
         regex=r"^(\+)[1-9][0-9\-\(\)\.]{9,15}$",
     )
     user_type: UserType
+    user_role: UserRole
```
Comment (sumit-rawat-quo) [inline backend/models/user.py]: This should be optional field. There can be a case where user decided to not select "is creating for someone else", in that case userRole field should be empty/blank.

File: backend/models/user.py [hunk 3]
```diff
@@ -3,35 +3,48 @@
...
     child: List[Child]
     relation_with_child: Optional[str]
+    is_created_for_someone_else: Optional[bool] = False
```
Comment (sumit-rawat-quo) [inline backend/models/user.py]: relation_with_child field is not required anymore if you are using user_role for that. You can remove that.
On the separate note, relation_with_child is more readable field so I would suggest you to rename "user_role" to "relation_with_child" . What do you think ?

File: backend/routes/user.py [hunk 4]
```diff
@@ -25,15 +25,27 @@ def otp_generate_save(phone_number):
...
+    # Check if the user wants to create an account for someone else
+    if user.is_created_for_someone_else:
+        # Get the name of the person for whom the account is being created
+        full_name = user.name
+
+        # Add the new user to the database with the specified information
+        user_dict.update({
+            'full_name': full_name
+        })
+
```
Comment (sumit-rawat-quo) [inline backend/routes/user.py]: Why we are saving 'full_name' field in db ? I am missing something here, can you please explain it to me. ?

File: backend/models/user.py (hunk 3 above)
Comment (sadam-123) [reply backend/models/user.py]: Yes I agree with you and  I will rename that

File: backend/routes/user.py (hunk 4 above)
Comment (sadam-123) [reply backend/routes/user.py]: Here what I assume that if  user select checkbox  for the someone else in CreateUserSchema model we have "name" field for user.So after select checkbox 'full_name': full_name is executed for checking correct user who are going to created account that full_name was saved in db.
 As per our requirements If full_name  filed and save it in db  is not necessary  I can remove that.

Comment (sumit-rawat-quo) [reply backend/routes/user.py]: I think already existing name field suffices the requirements. There is no requirement for saving full_name for whom we are creating the account. You can remove that.

Comment (sadam-123) [reply backend/routes/user.py]: Yes,here I removed Full_Name and  I have commited changes regarding that

File: backend/routes/user.py [hunk 5]
```diff
@@ -1,47 +1,76 @@
...
 
-    # check if same record exists
-    if database.user.find_one({'phone_number': user_dict['phone_number']}):
-        raise HTTPException(status_code=409, detail="Phone number already exists !")
+    # Check if the user wants to create an account for someone else
+    if user.is_created_for_someone_else:
+        user_dict.update({})
+
+    else:
+        # check if same record exists
```
Comment (sumit-rawat-quo) [inline backend/routes/user.py]: Do we need this if else block ? If not please remove this .

File: backend/routes/user.py [hunk 6]
```diff
@@ -39,11 +44,15 @@ async def create_account(user: CreateUserSchema= Body(...)):
...
 
-    # # generate and save otp for user
+    # generate and save otp for user
     otp = otp_generate_save(user_dict['phone_number'])
-    # # send otp to users phone number
+    # send otp to users phone number
     send_otp_to_phone(user_dict['phone_number'], otp)
 
-    return {'status_code':200, 'message': 'User saved successfully'}
+    return {'status_code': 400, 'message': 'OTP expired. Resend OTP?'}
```
Comment (sumit-rawat-quo) [inline backend/routes/user.py]: Please use this instead 
return {'status_code':200, 'message': 'User saved successfully'}

**Review by sumit-rawat-quo — APPROVED** [review]
LGTM

**Pull request merged**